### DISCUSSION TITLE
Howto attach image as metadata and generate link?

### DISCUSSION THREAD

**Discussion by jefdriesen** [getpelican/pelican#3133] [Q&A]
I'm trying to attach an image as article metadata:
```
title: My title
description: My description
photo: {attach}myphoto.jpg
```
And render it with a custom template:
```
<figure>
    <a href="{{ article.photo }}"><img src="{{ article.photo }}" alt="{{ article.description }}" /></a>
    <figcaption>{{ article.description }}</figcaption>
</figure>
```
But this does copy the text as-is, without generating a URL:
```
<figure>
    <a href="{attach}myphoto.jpg"><img src="{attach}myphoto.jpg" alt="My description" /></a>
    <figcaption>My description</figcaption>
</figure>
```
How can I make pelican generate a link, like it does when I use this construct in the content of the article:
```
[![My description]({attach}myphoto.jpg "My description")]({attach}myphoto.jpg)
```

**Comment by justinmayer** [comment]
I don't believe the `{attach}` syntax is available in metadata, or anywhere else except inside the main article/page content.

For one site I used the following metadata:

```
Image: anxious-girl
Image_ext: png
```

… and then in the theme's `article.html` template, I do things like:

```
<div class="image main"><img src="{{ SITEURL }}/images/{{ article.image|default(article.slug) }}-b.{{ article.image_ext }}" alt="{{ article.title|striptags }}" /></div>
```

**Comment by jefdriesen** [comment]
That works indeed for static images that are located in a fixed directory. But it looses the flexibility of the `{attach}` syntax to move the images to the correct output.

For example, when organizing posts in subdirectories (which is very practical for articles with many images):
```
content/blog/myarticle/myarticle.md
content/blog/myarticle/myphoto.jpg
```
but with different directory layout for the output:
```
ARTICLE_SAVE_AS = '{date:%Y}/{slug}.html'
ARTICLE_URL = '{date:%Y}/{slug}.html'
```
I would need to hardcode the correct path in the theme. With the  `{attach}` syntax the correct URL is always generated automatically, and the theme doesn't need to be aware of the settings.

**Reply by justinmayer** [reply]
Understood, but as I said, I don’t believe the `{attach}` syntax is currently supported in post metadata. Is that something you would like to work on implementing?

**Reply by jefdriesen** [reply]
My python experience is limited, so I'm not sure this is something I will be able to implement myself.

**Reply by justinmayer** [reply]
Yeah, I realize it would take some learning and work for you to implement such a feature, and that may not be worth the utility you would get from having that feature. Certainly understandable!
